.github/workflows/build-and-release.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and upload to PyPI
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, macos-13, windows-latest]
```

Comment:
Maybe we can set macos-14 explicitly like [this](https://github.com/Qiskit/qiskit/blob/dd145b54be9d583d3857edfbd81e910600249adf/.github/workflows/wheels.yml#L15)?

Reply:
I think it's better to just test the latest version, just as we do with Ubuntu and Windows. The reason to include `macos-13` is because it's the last Intel Mac runner. `macos-latest` will refer to an Apple Silicon runner.